compiler/rustc_lint/src/non_local_def.rs

```diff
@@ -115,23 +113,20 @@ impl<'tcx> LateLintPass<'tcx> for NonLocalDefinitions {
                 // We also ignore anon-const in item by including the anon-const
                 // parent as well; and since it's quite uncommon, we use smallvec
                 // to avoid unnecessary heap allocations.
-                let local_parents: SmallVec<[DefId; 1]> = if parent_def_kind == DefKind::Const
-                    && parent_opt_item_name == Some(kw::Underscore)
-                {
-                    smallvec![parent, cx.tcx.parent(parent)]
-                } else {
-                    smallvec![parent]
-                };
+                let parent_parent = (parent_def_kind == DefKind::Const
+                    && parent_opt_item_name == Some(kw::Underscore))
+                .then(|| cx.tcx.parent(parent));
```

Comment:
Please prefer an if/else block for readability.

Reply:
Yeah, rustfmt doesn't help here. Done.